static/js/src/tabbed-content.js

```diff
@@ -0,0 +1,133 @@
+      compatibleKeys = keys;
+      compatibleDirection = direction;
+      pressed = event.code;
+    }
+
+    if (compatibleDirection[pressed]) {
+      var target = event.target;
+      if (target.index !== undefined) {
+        if (tabs[target.index + compatibleDirection[pressed]]) {
+          tabs[target.index + compatibleDirection[pressed]].focus();
+        } else if (pressed === compatibleKeys.left) {
+          tabs[tabs.length - 1].focus();
+        } else if (pressed === compatibleKeys.right) {
+          tabs[0].focus();
+        }
+      }
+    }
+  }
+
+  /**
+      Attaches a number of events that each trigger
+      the reveal of the chosen tab content
+      @param {Array} tabs an array of tabs within a container
+    */
+  function attachEvents(tabs) {
+    tabs.forEach(function (tab, index) {
+      tab.addEventListener("keyup", function (e) {
+        var compatibleKeys = IEKeys;
+        var key = e.keyCode;
+
+        if (e.code) {
+          compatibleKeys = keys;
+          key = e.code;
+        }
+
+        if (key === compatibleKeys.left || key === compatibleKeys.right) {
+          switchTabOnArrowPress(e, tabs);
+        }
+      });
+
+      tab.addEventListener("click", function (e) {
+        e.preventDefault();
+        setActiveTab(tab, tabs);
+      });
+
+      tab.addEventListener("focus", function () {
+        setActiveTab(tab, tabs);
+      });
+
+      tab.index = index;
+    });
+  }
+
+  /**
+      Cycles through an array of tab elements and ensures 
+      only the target tab and its content are selected
+      @param {HTMLElement} tab the tab whose content will be shown
+      @param {Array} tabs an array of tabs within a container
+    */
+  function setActiveTab(tab, tabs) {
+    tabs.forEach(function (tabElement) {
+      var tabContent = document.querySelectorAll(
+        "#" + tabElement.getAttribute("aria-controls")
+      );
+      tabContent.forEach(function (content) {
+        if (tabElement === tab) {
+          tabElement.setAttribute("aria-selected", true);
+          content.removeAttribute("hidden");
+        } else {
+          tabElement.setAttribute("aria-selected", false);
+          content.setAttribute("hidden", true);
+        }
+      });
+    });
+  }
+
+  /**
+      Attaches events to tab links within a given parent element,
+      and sets the active tab if the current hash matches the id
+      of an element controlled by a tab link
+      @param {String} selector class name of the element 
+      containing the tabs we want to attach events to
+    */
+  function initTabs(selector) {
+    var tabContainers = [].slice.call(document.querySelectorAll(selector));
+
+    tabContainers.forEach(function (tabContainer) {
+      var tabs = [].slice.call(
+        tabContainer.querySelectorAll("[aria-controls]")
+      );
+      attachEvents(tabs);
+    });
+  }
+
+  document.addEventListener("DOMContentLoaded", function () {
+    initTabs('[role="tablist"]');
+  });
```

Comment:
Good idea, would it not be sufficient to just create a custom role class that we want the script to apply to?

Reply:
`role` attributes are specific to [WAI-ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles), and have set values - we just need a class to the element i.e. `js-tabbed-content` and target that instead.

Reply:
Oh - I see what you mean after re-reading. The `initTabs` function takes an argument so that we have the ability to initialise it on a given instance of the tabbed content pattern, so I think it's cleaner to include the script and explicitly call `initTab` in the template, sort of like we do with latest news: https://github.com/canonical-web-and-design/ubuntu.com/blob/main/templates/shared/contextual_footers/_ai_further_reading.html#L14

Reply:
Ah ok, makes sense. But what is the benefit of an explicit call to``initTab``, apart from making the code more readable? It makes sense with 'latest news' as there are different arguments so different use cases, but there is only one use for this script.

Reply:
That's a fair point! OK, leave it there, but with the new class.